### ISSUE TITLE
Convert factor to Logical gives an error

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#5869]
I think this is all linked to our use of labelled data?

The example I found was with the Dodoma data in the R-Instat climatic library.  I made a logical variable as in (tmax > 29).  Saved into tmax29.  Then I duplicated the variable into a factor with 2 levels.  This was called tmax29F.  All was fine.

Then I tried to make it into a logical variable again.

a) Using Prepare > Data Frame > Convert doesn't work.  It says:

![image](https://user-images.githubusercontent.com/11226469/85945591-d4ee9700-b936-11ea-8aad-c9c85eb9ae8a.png)

b) Using Right-click and Duplicate Column ( or Prepare > Column: Calculate > Duplicate) gives the same error.

c) Using the calculator with as.logical(tmax29F) works fine.

So I propose this is all another part of our labelling of the variables, which also still causes problems elsewhere.  

I suggest it is time to fix this - and perhaps more generally?   I assume it is all related to labelling variables and hence perhaps #5465 could finally be sort out at the same time?

**Comment by lilyclements** [comment]
So I've had a little look and to simplify this problem I tend to get this when converting any factor to a logical variable (via this dialog, at least).

I am not confident with the `data_object_R6` file, however, I assume the change should be made on line 1209:

```
   else if(to_type == "logical") {
      if(is.logical.like(curr_col)) new_col <- as.logical(curr_col)
      else stop("Column is not numeric or contains values other than 0 and 1. Converting to logical would result in losing information.")
    }
```

Base-R will convert a factor to a logical if that factor has levels "TRUE/FALSE". Otherwise, it makes them missing (see below).
Would we want R-Instat to follow suit (or to work for other two-level factors such as 0/1)?

```
> x <- c(rep("0", 3), rep("1", 3))
> as.logical(as.factor(x))
[1] NA NA NA NA NA NA
> x <- c(rep("TRUE", 3), rep("FALSE", 3))
> as.logical(as.factor(x))
[1]  TRUE  TRUE  TRUE FALSE FALSE FALSE
```

**Comment by dannyparsons** [comment]
Yes this is a good solution. I suggest we have an equivalent check on factor columns as on numeric - if all the levels are `TRUE`, `FALSE` then convert, if these are not the only levels then stop.

The slightly longer term solution is to be able to have a warning message if NAs will be introduced to allow users the option to convert even if some values will be lost.

**Comment by lilyclements** [comment]
@dannyparsons on line 1209 would we want something like

```
if (!any(as.logical(curr_col) %in% c(TRUE, FALSE)) == FALSE) new_col <- as.logical(curr_col)
else stop("Column is not numeric or contains values other than 0 and 1. Converting to logical would result in losing information.")
```

The `else` statement would need to change though since this does convert numeric to logicals, even if they are not logical-like, just as in R:

```
x <- c(rep(5, 3), rep(1, 3))
x
is.logical.like(x)
as.logical(x)
```
